### DISCUSSION TITLE
Performance and interface improvements

### DISCUSSION THREAD

**Discussion by ghost** [lukehaas/RunJS#599] [Q&A]
1. Are there any sketches or thoughts about changing the runtime behind Runjs? Following trends for Bun or Deno.
2. Is there also the possibility of having a sidebar for organizing files and folders in the future?

**Comment by lukehaas** [comment]
RunJS is built on Electron, which is where the dependence on Node is inherited from. For RunJS to switch to Deno or Bun, Electron first needs to make the switch. You can find a discussion on Electron moving to Deno here: https://github.com/electron/electron/issues/23613

Adding a sidebar for files and folders would push RunJS more in the direction of an IDE or some kind of editor for building apps. This is not what RunJS is trying to be. There are many great tools that already do that, such as VSCode. RunJS is intended to be a playground for tinkering and experimenting etc.

**Reply by ghost** [reply]
Could you consider a fork with a sidebar? I'm sure many people would be interested in obtaining the result of two tools in one. Playground and Snippet Manager.

I just love Runjs! I miss a sidebar to organize files, which forces me to use a second tool like SnipperApp or SnippetsLabs.

**Reply by lukehaas** [reply]
I thought you were asking for a sidebar to manage files, not snippets?
RunJS has a snippet manager already.
![snippets](https://github.com/lukehaas/RunJS/assets/741344/59a26b60-941f-4fa1-a075-546ddc93449f)
